.pre-commit-config.yaml

```diff
@@ -0,0 +1,46 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.0.1
+  hooks:
+  - id: check-added-large-files
+  - id: check-case-conflict
+  - id: check-yaml
+  - id: debug-statements
+  - id: end-of-file-fixer
+    exclude: ".*(data.*|extern.*|licenses.*|_parsetab.py)$"
+  - id: trailing-whitespace
+    exclude: ".*(data.*|extern.*|licenses.*|_parsetab.py)$"
+# - repo: https://github.com/timothycrosley/isort
+#   rev: 5.9.2
+#   hooks:
+#   - id: isort
+# - repo: https://github.com/asottile/pyupgrade
+#   rev: v2.21.0
+#   hooks:
+#     - id: pyupgrade
```

Comment:
2 additional checks we could use: 
- we have an isort config, should we run it (I thought we ran it once in the past but actually it seems not)
- pyupgrade was used to convert to fstrings in #8958, it can also provide various little improvements (https://github.com/asottile/pyupgrade) using syntax from the recent versions of Python.

Reply:
Other good built-in ones (v3.4.0>) are:

check-added-large-files
check-case-conflict
check-merge-conflict
check-symlinks
check-toml
check-yaml
debug-statements
destroyed-symlinks
detect-private-key
end-of-file-fixer
fix-encoding-pragma
requirements-txt-fixer
trailing-whitespace

Reply:
> isort config, should we run it

👍

Reply:
And https://github.com/pre-commit/pygrep-hooks is great for rst checks.

- rst-backticks
- rst-directive-colons
- rst-inline-touching-normal

I also use it for checking that I give the right error message in ``noqa``  e.g. ``noqa: F403``

- python-check-blanket-noqa

Reply:
> Other good built-in ones (v3.4.0>) are:

I already added most of them:
```
  - id: check-added-large-files
  - id: check-case-conflict
  - id: check-yaml
  - id: debug-statements
  - id: end-of-file-fixer
  - id: trailing-whitespace
```

Reply:
Looks like they are working! The CI has found oodles of corrections in the code.

Reply:
Is "oodles" good or bad? 😅

Reply:
Both! 😆